contracts/deploy/003_assignWinManager.ts

```diff
@@ -0,0 +1,32 @@
+import "hardhat-deploy";
+import { HardhatRuntimeEnvironment } from "hardhat/types";
+import { DeployFunction } from "hardhat-deploy/types";
+
+const func: DeployFunction = async function ({
+    deployments,
+    getNamedAccounts
+}: HardhatRuntimeEnvironment) {
+
+    const { get, read, execute } = deployments;
+    const { deployer } = await getNamedAccounts();
+
+    const battle = await get("Battle");
+
+    await execute(
```

Comment:
this will happen every time, so needs to be wrapped in a if (battle.newlyDeployed)

Reply:
Understood. Will push a new commit for this.